### PR TITLE
Add globalization

### PR DESCRIPTION
**Pull request by superkhau**
Connect to strongloop/loopback/issues/2422

### CODE REVIEW

**Comment by superkhau** [comment]
@slnode test please

**Comment by 0candy** [comment]
LGTM

File: example/hidden.js [hunk 1]
```diff
@@ -20,6 +22,6 @@ app.model(User);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at localhost:' + port + '/explorer');
+console.log(g.f('Explorer mounted at localhost:%s/explorer', port));
```
Comment (0candy) [inline example/hidden.js]: Should add `{{}}` around the url

File: example/simple.js [hunk 2]
```diff
@@ -19,6 +21,6 @@ app.model(Product);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at http://localhost:' + port + '/explorer');
+console.log(g.f('Explorer mounted at http://localhost:%s/explorer', port));
```
Comment (0candy) [inline example/simple.js]: Add `{{}}` around url

File: index.js [hunk 3]
```diff
@@ -38,7 +41,7 @@ function routes(loopbackApplication, options) {
   loopback.version.split('.')[0] || 1;
 
   if (loopbackMajor < 2) {
-    throw new Error('loopback-component-explorer requires loopback 2.0 or newer');
+    throw new Error(g.t('loopback-component-explorer requires loopback 2.0 or newer'));
```
Comment (0candy) [inline index.js]: Add `{{}}` around `loopback-component-explorer` and `loopback`

**Comment by superkhau** [comment]
@0candy Updated, review again pls.

File: example/hidden.js [hunk 4]
```diff
@@ -20,6 +22,6 @@ app.model(User);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at localhost:' + port + '/explorer');
+console.log(g.f('Explorer mounted at {{localhost:%s/explorer}}', port));
```
Comment (0candy) [inline example/hidden.js]: You have to run slt-globalize -e again for intl/en/messages.json to get updated with the brackets change.

**Comment by superkhau** [comment]
@0candy Updated, PTAL again.

**Comment by 0candy** [comment]
Please make sure `var SG = require('strong-globalize');` is at the beginning of the file.

**Comment by superkhau** [comment]
@slnode test please

**Comment by superkhau** [comment]
@Amir-61 @loay PTAL

**Comment by superkhau** [comment]
Actually sorry @Amir-61 @loay, I looked up the wrong repo owners.

@gunjpan @richardpringle PTAL instead. :muscle:

File: example/hidden.js (hunk 4 above)
Comment (richardpringle) [inline example/hidden.js]: I don't think Explorer should be translated, should it?

File: example/simple.js [hunk 5]
```diff
@@ -19,6 +21,6 @@ app.model(Product);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at http://localhost:' + port + '/explorer');
+console.log(g.f('Explorer mounted at {{http://localhost:%s/explorer}}', port));
```
Comment (richardpringle) [inline example/simple.js]: same comment as before,

I think it should be `{{Explorer}}`.

**Comment by richardpringle** [comment]
@superkhau, left a couple comments.

**Comment by superkhau** [comment]
@gunjpan @richardpringle Fixed up based on feedback. PTAL again.

File: example/hidden.js [hunk 6]
```diff
@@ -20,6 +22,6 @@ app.model(User);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at localhost:' + port + '/explorer');
+g.log('{{Explorer}} mounted at {{localhost:%s/explorer}}', port);
```
Comment (richardpringle) [inline example/hidden.js]: :+1: 

@0candy, it was said somewhere that words with an uppercase letter to not get translated, does that include words at the beginning of a sentence?

File: example/simple.js [hunk 7]
```diff
@@ -19,6 +21,6 @@ app.model(Product);
 var apiPath = '/api';
 explorer(app, { basePath: apiPath });
 app.use(apiPath, loopback.rest());
-console.log('Explorer mounted at http://localhost:' + port + '/explorer');
+g.log('{{Explorer}} mounted at {{http://localhost:%s/explorer}}', port);
```
Comment (richardpringle) [inline example/simple.js]: :+1:

File: intl/en/messages.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "b9e94c12da3208f46a969191874c425c": "{{Explorer}} mounted at {{localhost:{0}/explorer}}",
+  "88bef790f515ec7b7af0c6871e60c077": "{{Explorer}} mounted at {{http://localhost:{0}/explorer}}",
+  "f3f2b04c273e23780d76306f8c72a60f": "{{loopback-component-explorer}} requires {{loopback}} 2.0 or newer"
```
Comment (richardpringle) [inline intl/en/messages.json]: :+1:  :+1:  :+1:

File: example/hidden.js (hunk 6 above)
Comment (0candy) [inline example/hidden.js]: @richardpringle It is only for words with ALL caps. ie. `API` that don't get translated or any words you put within the double brace. In your case, both `Explorer` and the url will not be translated.

**Comment by superkhau** [comment]
@richardpringle Please leave LGTM next time. I assume all the thumbs ups means :ship:?

**Pull request merged**